src/containers/App.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import './App.css';
+import MovieTile from '../components/MovieTile/MovieTile';
+import SearchForm from '../components/SearchForm/SearchForm';
+import WhichPage from '../components/WhichPage/WhichPage';
+import ButtonPageControl from '../components/ButtonPageControl/ButtonPageControl';
+import MovieDetail from '../components/MovieDetail/MovieDetail';
+
+import {BrowserRouter as Router, Switch, Route} from 'react-router-dom';
+
+
+
+
+
+function App() {
+    const [movies, setMovies] = useState(false);
+    const [pages, setPages] = useState(false);
+    const [currentQuery, setCurrentQuery] = useState('');
+    const [currentPage, setCurrentPage] = useState(1);
+    const [isLoading, setIsLoading] = useState(false);
+    const [connectionError, setConnectionError] = useState(false);
+
+    const key = 'ae56d5e33eecc34a48f563c98dd330ad';
+
+    useEffect(() => {
+      if (currentQuery !== '') {
+        setIsLoading(true)
+      fetch(`https://api.themoviedb.org/3/search/movie?api_key=${key}&language=en-EN&query=${currentQuery}&page=${currentPage}&include_adult=false`)
+        .then(res => res.json())
+        .then(data => {
+          setMovies(data.results)
+          setPages(data.total_pages)
+          setIsLoading(false)
+        })
+        .catch(err => setConnectionError(err))
+      } else {
+        setMovies(false)
+        setPages(false)
+      }
```

Comment:
https://learningactors.com/javascript-guard-clauses-how-you-can-refactor-conditional-logic/

przydatny guard pattern - pomaga zapobiec wielu zagnieżdżonym ifom

Reply:
Nie do końca rozumiem.

Na tym przykładzie, przy instrukcjach if/else mam wywoływać funkcje z zawartością która teraz jest w blokach?

Reply:
idea jest taka, że jak mamy akcje które mają być wykonywane w zależności od jakiegoś warunku
np 
```js
if (hungry) {
goEatSomething()
} else {
continueWatchTV()
}

to korzystając z guard pattern likwidujemy trochę skomplikowania i zagnieżdżeń

if (hungry) {
return goEatSomething()
}

return continueWatchingTV()

```
najbardziej jest to widoczne na np zagnieżdżonych ifach

```js
if (hungry) {
  if (glutenIntolerant) {
    eatVegetable()
  } else {
    makeASandwitch()
  }
} else {
continueWatchTV()
} 

można zamienić na:

if (!hungry) {
return continueWatchTV()
}
if (!glutenIntolerant) {
return makeASandwitch()
} 
return eatAVegetable()

Reply:
korzystamy tutaj z przerywania wywołania funkcji po return żeby uprościć kod :)